dot/parachain/network-bridge/messages/overseer_messages.go

```diff
@@ -0,0 +1,63 @@
+package messages
+
+import (
+	collatorprotocolmessages "github.com/ChainSafe/gossamer/dot/parachain/collator-protocol/messages"
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	validationprotocol "github.com/ChainSafe/gossamer/dot/parachain/validation-protocol"
+
+	"github.com/ChainSafe/gossamer/dot/peerset"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+// NOTE: This is not same as corresponding rust structure
+// TODO: If need be, add ability to report multiple peers in batches
+type ReportPeer struct {
+	PeerID           peer.ID
+	ReputationChange peerset.ReputationChange
+}
```

Comment:
What is different in the Rust code? Can you add more details?

Reply:
```
/// Type of peer reporting
#[derive(Debug)]
pub enum ReportPeerMessage {
	/// Single peer report about malicious actions which should be sent right away
	Single(PeerId, ReputationChange),
	/// Delayed report for other actions.
	Batch(HashMap<PeerId, i32>),
}
```
This is the rust struct, but I don't think we need to add that. The comment is just meant to help us better understand this. I am going to remove the NOTE here, since it could be confusing.